llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -339,12 +374,23 @@ def _is_last_batch(self, state: State):
 
         return False
 
-    def _all_child_processes_done(self) -> bool:
-        not_done = any(process.is_alive() for process in self.child_processes)
+    def _all_register_processes_done(self) -> bool:
+        not_done = any(
+            process.is_alive() for process in self.register_processes
+        )
         x = torch.tensor(1 if not_done else 0).to(device='cuda')
         dist.all_reduce(x, reduce_operation='MAX')
         return x.item() == 0
 
+    def _any_register_processes_error(self) -> bool:
+        has_errors = any(
+            process.exitcode is not None and process.exitcode != 0
+            for process in self.register_processes
+        )
+        x = torch.tensor(1 if has_errors else 0).to(device='cuda')
```

Comment:
mildly unfortunate this only works with gpu. would be better to get the device from state. up to you if you want to do or not since it was like this before your pr.

Reply:
done